code/game/objects/items/tanks/watertank.dm

```diff
@@ -140,6 +140,10 @@
 		return INITIALIZE_HINT_QDEL
 	reagents = tank.reagents //This mister is really just a proxy for the tank's reagents
 
+/obj/item/reagent_containers/spray/mister/Destroy(force)
+	tank = null
```

Comment:
same here

Reply:
This is a funky case, because the tank's destroy already QDEL_NULL's the mister. Is a cyclic QDEL_NULL safe?

Reply:
check for !QDELETED(x)

Reply:
Sounds good, done.

Reply:
Update: Actually realized that there was no reason for the tank var to be a class variable, so I completely removed it, as the type wasn't reliably correct anyway and it served no function.